File: notes-n-resources/Data-Structures-N-Algo/_DS-n-Algos/Interview-Problems/DataStructures/Stack/Python/arrayStack.py
# Stack implementation using Array in Python
# Stack Methods : push, pop, peek, isEmpty

from sys import maxsize


def initializeStack():
    stack = []
    return stack


def isEmpty(stack):
    if len(stack) == 0:
        return True
    else:
        return False


def pop(stack):
    if isEmpty(stack):
        return str(-maxsize - 1)
    return stack.pop()


def push(item, stack):
    stack.append(item)
    print(str(item) + " pushed into stack")


def peek(stack):
    if isEmpty(stack):
        return str(-maxsize - 1)
    return stack[len(stack) - 1]


# Driver code to test the functions
stack = initializeStack()
push(3, stack)
push(4, stack)
push(5, stack)
print(peek(stack))
print(pop(stack))

File: notes-n-resources/Data-Structures-N-Algo/_DS-n-Algos/Interview-Problems/DataStructures/Stack/Java/ArrayStack.java
public class ArrayStack {
  int stack[] = new int[5];
  int t = 0;

  public void push(int data) {
    if (t == 5) {
      System.out.println("Stack is full!");
    } else {
      stack[t] = data;
      t++;
    }
  }

  public int size() {
    return t;
  }

  public void printStack() {
    for (int n : stack) {
      System.out.print(n + "\t");
    }
  }

  public int pop() {
    if (isEmpty()) {
      System.out.println("Stack is Empty!");
      return 0;
    } else {
      t--;
      int popped = stack[t];
      stack[t] = 0;
      return popped;
    }
  }

  public int top() {
    return stack[t];
  }

  public boolean isEmpty() {
    if (t == 0) {
      return true;
    } else {
      return false;
    }
  }
}
